### ISSUE TITLE
Entities Card - Header toggle on by default but not showing

### ISSUE THREAD

**Issue by Xitee1** [home-assistant/frontend#17475]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The `entities` card has to option to show the switch for all devices in the header.
This option is on by default after you've add the card and click on edit afterwards.
However the switch is not shown in the header. You have to turn the option off and on again to make the switch in the header show up.
![chrome_21dsEQTrl0](https://github.com/home-assistant/frontend/assets/59659167/1c064fd7-78f2-4b16-8561-651c368796d0)
![chrome_DWWbvv53qT](https://github.com/home-assistant/frontend/assets/59659167/9a8bc52d-efa9-42b6-9151-88ea278c00dc)

### Describe the behavior you expected

Probably best would be to default that option to off.

### Steps to reproduce the issue

1. Add new `entities` type card with 2 switches.
2. Click on edit.

### What version of Home Assistant Core has the issue?

core-2023.7.3

### Additional information

The problem probably is that if the option isn't specified in the YAML configuration, it defaults to false.
But the default of the visual editor is that this option is on.

We need to match them by either turning the switch off by default in the visual editor or defaulting to true if the option isn't defined in the YAML configuration. I would prefer the first option.

**Comment by ildar170975** [comment]
In English:

![image](https://github.com/home-assistant/frontend/assets/71872483/5b99b536-a58b-4533-85ee-cabed25aff70)

The "header toggle" is enabled by default; but the title is not defined by default.

When a user defines a title - then this toggle is shown:

![image](https://github.com/home-assistant/frontend/assets/71872483/dc27ac6b-e43a-4fa5-a8f2-9500226e2ab7)


> You have to turn the option off and on again to make the switch in the header show up.

IMHO when the title is not defined - the header toggle should NOT be shown at all:

![image](https://github.com/home-assistant/frontend/assets/71872483/b8a48726-24fd-462b-8f79-a2363886261e)

But may be someone really using this header toggle w/o a title....

**Comment by Xitee1** [comment]
Yes, It would probably be correct that the header switch should not show up when there is no title.

> IMHO when the title is not defined - the header toggle should NOT be shown at all:

Who knows if someone is using it this way. Only allowing it to show when there's a title would break that functionality.
But yes it is really ugly this way. And if someone really wants to use it like so, they can still add a `title: ''`.

But the second problem is that it might be confusing when someone has not set a title, clicks the switch and nothing happens to the card.

So best thing we can probably do is to check if there's an title and only show the header with switch if so.
When there's no title defined the switch needs to be grayed out to avoid confusion.

**Comment by ildar170975** [comment]
> when someone has not set a title, clicks the switch and nothing happens to the card.

Agree, could be confusing for beginners.



> When there's no title defined the switch needs to be grayed out to avoid confusion.

and probably set to OFF.

**Comment by karwosts** [comment]
The issue seems to be that we have three different behaviors (toggle on, toggle off, and toggle undefined) that are represented by a single boolean selector that can only represent two states. 

The way I can imagine solving this in a backwards compatible way that doesn't take away any functionality would be:

If show_header_toggle is defined in the yaml, render the selector to match the yaml. 

If show_header_toggle is undefined in yaml:
 - if a title is defined, render the toggle in the UI as `on`. Clicking the toggle will turn it off, write `show_header_toggle: false` to the yaml, and remove the visible toggle in the card preview. 
 - if a title is not defined, render the toggle as `off`. Clicking the toggle will turn it on, write `show_header_toggle: true` to the yaml, and add a visible toggle in the card preview. 


With that I think in all cases a single click of the toggle would have the desired effect.

**Comment by ildar170975** [comment]
up

**Comment by Xitee1** [comment]
Still an issue.

**Comment by ildar170975** [comment]
up

**Comment by Xitee1** [comment]
Still an issue

**Comment by Xitee1** [comment]
Didn't check but probably still an issue
